supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -93,14 +97,29 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
     case ExecutionResultStored(hash) => log.debug("{} Cache entry was stored for Job with hash {}.", tag, hash)
 
     case ExecutionResultAlreadyExist => log.warning("{} Cache entry for hash {} already exist.", tag, jobHash)
+
+    case KvPair(key, Some(jobId)) if key.key == HtCondorJobIdKey =>
+      log.debug("{} Found job id {} for job {} trying to recover job now.", tag, jobId, jobDescriptor.key)
+      executionResponse success trackTaskToCompletion(jobId)
+
+    case KvKeyLookupFailed(_) =>
+      log.debug("{} Job id not found for job {} falling back to execute.", tag, jobDescriptor.key)
```

Comment:
Also, just logging `jobDescriptor.key.tag` should be enough here.. as opposed to the structure that you're logging now (`jobDescriptor.key`), which IMO is a little noisy and redundant.
(Same for logging errors below)